File: app-server/src/test/java/com/github/arhor/simple/expense/tracker/ContextTest.java
package com.github.arhor.simple.expense.tracker;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.ApplicationContext;
import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;

import static org.assertj.core.api.Assertions.assertThat;

@SpringJUnitConfig(ContextTest.EmptyContextConfig.class)
class ContextTest {

    @TestConfiguration(proxyBeanMethods = false)
    static class EmptyContextConfig {
    }

    @Autowired
    private ApplicationContext ctx;

    @Test
    void application_context_should_be_loaded_successfully() {
        assertThat(ctx)
            .isNotNull();
    }
}

File: app-server/src/test/kotlin/com/github/arhor/simple/expense/tracker/ContextTest.kt
package com.github.arhor.simple.expense.tracker

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.ApplicationContext
import org.springframework.test.context.junit.jupiter.SpringJUnitConfig

@SpringJUnitConfig(ContextTest.EmptyContextConfig::class)
internal class ContextTest {

    @TestConfiguration(proxyBeanMethods = false)
    class EmptyContextConfig

    @Autowired
    private lateinit var ctx: ApplicationContext

    @Test
    fun `application context should be loaded successfully`() {
        assertThat(ctx)
            .isNotNull
    }
}
